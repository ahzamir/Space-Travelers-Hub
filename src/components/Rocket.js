import { useDispatch } from 'react-redux/es/exports';
import { addReservation } from '../redux-states/rockets/RocketState';

function Rocket(props) {
  const dispatch = useDispatch();
  const list = props;
  const {
    id, name, image, description, reserved,
  } = list.myRockets;

  const reservationHandler = () => {
    dispatch(addReservation(id));
  };

  return (
    <div className="rocket">
      <img src={image} alt={name} className="rocket-images" />
      <div className="rocket-info">
        <span className="rocket-name">
          {name}
        </span>
        <div className="rocket-reserv">
          <div className="rocket-bagde" style={reserved ? null : { display: 'none' }}>Reserved</div>
          <span className="rocket-description">
            {' '}
            {description}
            {' '}
          </span>
        </div>
        { reserved ? <button type="button" className="cancel-button" onClick={reservationHandler}>Cancel Reservation</button>
          : <button type="button" className="reserve-button" onClick={reservationHandler}>Reserve Rocket</button> }
      </div>
    </div>
  );
}

export default Rocket;
