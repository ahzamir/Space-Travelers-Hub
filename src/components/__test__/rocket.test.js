import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux-states/configureStore';

import Rocket from '../rockets/Rockets';

describe('Testing Mission component', () => {
  render(
    <Provider store={store}>
      <Rocket />
    </Provider>,
  );
  test('Testing the dom for success rendering elements on Mission component', () => {
    expect(screen.queryByTestId('rocketid')).toBeInTheDocument();
  });

  test('Testing the dom for non existing elemente on the page after rendering', () => {
    expect(screen.queryByTestId('nothing')).not.toBeInTheDocument();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rocket page="just for test">Missions</Rocket>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
