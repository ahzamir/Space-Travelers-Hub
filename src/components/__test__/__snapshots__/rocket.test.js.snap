// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Mission component renders correctly 1`] = `
<div
  className="rockets-container"
  data-testid="rocketid"
>
  <ul
    className="rockets"
  />
</div>
`;
